File: Notebook.java

public class Notebook extends Item{

    //Variables

    private int sections ;
    
    
    
    Notebook(String name,double price , String description , int stock , int id,int sections) throws IllegalCharacterException{

        super(name , price , description , stock , id);     //calls Item constructor
        
        this.sections = sections ;

    }



    //Methods

//-----------------getDetails implementation-----------------

    public String getDetails(){

        return ", Sections: " + Integer.toString(sections);

    }


//-------------------Setters-----------------------------------


    public void setSections(int s){

        this.sections = s ;

    }


//----------------Getters--------------------------------------

    public int getSections(){

        return this.sections ;

    }


}

File: C++/Notebook.cpp
#include "Notebook.h"

Notebook::Notebook(string n, double p, string d, int s, int i, int sec): Item(n, p, d, s, i), sections(sec) {

}

string Notebook::getDetails() {
    return ", Sections: " + to_string(this->sections);
}

string Notebook::getCategory() {
    return "notebook";
}

void Notebook::setSections(int s) {
    this->sections = s;
}

int Notebook::getSections() {
    return this->sections;
}
